pandas/core/window/rolling.py

```diff
@@ -473,12 +504,31 @@ def _apply(
 
                 def calc(x):
                     x = np.concatenate((x, additional_nans))
-                    min_periods = calculate_min_periods(
-                        window, self.min_periods, len(x), require_min_periods, floor
+                    if not isinstance(window, BaseIndexer):
+                        min_periods = calculate_min_periods(
```

Comment:
can you do these outside of the function? (as they don't depend on anything inside) and makes logic simpler

Reply:
These both depend on `x` after the `nan`s are appended from the `center` calculation.
`x = np.concatenate((x, additional_nans))`